server.go

```diff
@@ -814,6 +814,34 @@ func (s *Server) CopyMetastore(w io.Writer) error {
 	})
 }
 
+// CopyShard writes the requested shard to a writer.
+func (s *Server) CopyShard(w io.Writer, shardID uint64) error {
+	s.mu.RLock()
+
+	// Locate the shard.
+	sh, ok := s.shards[shardID]
+	if !ok {
+		s.mu.RUnlock()
+		return ErrShardNotFound
+	}
+	if sh.store == nil {
+		s.mu.RUnlock()
+		return ErrShardNotLocal
+	}
+
+	return sh.view(func(tx *shardtx) error {
+		s.mu.RUnlock() // Unlock so not held during long read.
+
+		// Set content length if this is a HTTP connection.
+		if w, ok := w.(http.ResponseWriter); ok {
+			w.Header().Set("Content-Length", strconv.Itoa(int(tx.Size())))
+		}
+
+		// Write entire shard to the writer.
+		return tx.Copy(w)
```

Comment:
Per https://godoc.org/github.com/boltdb/bolt#Tx.Copy, look like `WriteTo` might be preferred now?  You could also verify that actually  `tx.Size()` was written too.